### PR TITLE
Fix command handler cancellation

### LINKED ISSUE #24155: FormatCommandHandler.ReturnKey.cs is not cancellable, but doesn't disable cancellation
When formatting on Return FormatCommandHandler passes CancellationToken.None:
https://github.com/dotnet/roslyn/blob/d4dab355b96955aca5b4b0ebf6282575fad78ba8/src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.ReturnKey.cs#L18

It should either explicitly disable cancellation or use cancellation token passed via CommandExecutionContext (in the new commanding).

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Fixes #24155, which means if you click the cancel button during this blocking UI operation it'll actually do something. Also refactors the code to be less terrible.

<details><summary>Ask Mode template</summary>

### Customer scenario

The user tries to click the cancel button on a wait dialog, and discovers that it doesn't do anything.

### Bugs this fixes

#24155

### Workarounds, if any

None.

### Risk

Low; just passing through a cancellation token that was missed in another path.

### Performance impact

No change to user experience, but if our perf is terrible we won't be even more terrible by not letting th user cancel.

### Is this a regression from a previous update?

No.

### Root cause analysis

Simple oversight in code. By the looks of it, this was just refactored and not refactored well.

### How was the bug found?

By @olegtk has he was working in this area.

</details>

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
Note to reviewers: commit-at-a-time will be a lot easier to follow here. This code was 10% actual fixing, and 90% moving stuff around so I could know the 10% was remotely right.

File: src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs [hunk 1]
```diff
@@ -137,24 +132,42 @@ public void ExecuteReturnOrTypeCommand(EditorCommandArgs args, Action nextHandle
                 return;
             }
 
+            IList<TextChange> textChanges;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs]: i believe F#/TS go through these paths.  So might be worth manual validation of those guys.

File: src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs [hunk 2]
```diff
@@ -176,5 +189,10 @@ public void ExecuteReturnOrTypeCommand(EditorCommandArgs args, Action nextHandle
             // caret has moved to wrong position, move it back to correct position
             args.TextView.TryMoveCaretToAndEnsureVisible(oldCaretPosition);
         }
+
+        private CaretPreservingEditTransaction CreateEditTransaction(ITextView view, string description)
+        {
+            return new CaretPreservingEditTransaction(description, view, _undoHistoryRegistry, _editorOperationsFactoryService);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs]: consider =>

**Comment by jasonmalinowski** [comment]
@dotnet-bot retest this please.

**Comment by jasonmalinowski** [comment]
(the previous retest is to check the belief that @olegtk's change might conflict with mine, so re-checking for sanity.)

**Comment by jasonmalinowski** [comment]
Tagging @jinujoseph and @Pilchie for ask mode approval now.

**Comment by Pilchie** [comment]
Approved.

**Pull request merged**